src/wallet/receive.h

```diff
@@ -10,11 +10,55 @@
 #include <wallet/transaction.h>
 #include <wallet/wallet.h>
 
+isminetype InputIsMine(const CWallet& wallet, const CTxIn& txin) EXCLUSIVE_LOCKS_REQUIRED(wallet.cs_wallet);
+
+/** Returns whether all of the inputs match the filter */
+bool AllInputsMine(const CWallet& wallet, const CTransaction& tx, const isminefilter& filter);
+
+CAmount OutputGetCredit(const CWallet& wallet, const CTxOut& txout, const isminefilter& filter);
+CAmount TxGetCredit(const CWallet& wallet, const CTransaction& tx, const isminefilter& filter);
+
+bool ScriptIsChange(const CWallet& wallet, const CScript& script) EXCLUSIVE_LOCKS_REQUIRED(wallet.cs_wallet);
+bool OutputIsChange(const CWallet& wallet, const CTxOut& txout) EXCLUSIVE_LOCKS_REQUIRED(wallet.cs_wallet);
+CAmount OutputGetChange(const CWallet& wallet, const CTxOut& txout) EXCLUSIVE_LOCKS_REQUIRED(wallet.cs_wallet);
+CAmount TxGetChange(const CWallet& wallet, const CTransaction& tx);
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/21206#discussion_r625576746

> No need to export `OutputGetChange` and `TxGetChange` as they are not used outside of `receive.cpp`

I think in the future it might be good to move low-level code using `IsChange` logic inside `receive.cpp` so `IsChange` functions could be private. But for `GetChange` it seems most consistent to try to keeping exposing `Get{Credit,Debit,Change}` functions as a group. The fact that these two `GetChange` functions don't have callers I think is more a reflection of poor unit test coverage than that change amounts should be a private part of transaction accounting.